Replacing png icons

jaerrib opened the pull request:
Closes #21 by replacing the last remaining png files with svg variants (updated based on reviewer feedback on PR #23)

jaerrib committed:
Replace png version of nm_device_wired with svg version

jaerrib committed:
Update computer icon

jaerrib committed:
Replace input-joystick

jaerrib committed:
Replace media-flash-usb variations

jaerrib committed:
Replace kpager

jaerrib committed:
Replace proxy

jaerrib committed:
Update overrides after replacing computer icon with svg

jaerrib committed:
Improve contrast on darker taskbar

jaerrib committed:
Adjust to fit overall color theme

q4osteam commented:
This update looks better, however it would need some tuning still.

- computer.svg icon grey color may be the brighter one - as used by folder. We would guess the icon will pass the overall theme better, see the screenshot.
![VirtualBox_q4os-5](https://github.com/user-attachments/assets/3f6c9bbf-2f40-4b74-ae40-e04f1ea72c82)

- nm_device_wired.svg has got proper colors, but now it's distorted somehow. So it doesn't look as good as the previous png variant, see the screenshosts
![snapshot5](https://github.com/user-attachments/assets/1a3d8515-0eca-483e-8c83-7c074de4ff68)
![snapshot6](https://github.com/user-attachments/assets/0e908153-2a3c-45d7-956e-c943c1194636)

jaerrib committed:
Merge remote-tracking branch 'origin/development' into replacing-png-icons

jaerrib committed:
Update colors to match folders better

jaerrib committed:
Adjust node positions to correct blurring issue

jaerrib commented:
I updated the base color of the computer icon to be more inline with folders and replaced the blue screen with a dark one to keep a good contrast.
![computer](https://github.com/user-attachments/assets/f1649296-4c47-4874-b171-05598e8d8571)

With the nm_wired_device icon, I believe the blurring was linked to the relative positioning of the nodes. Adjusting some of the locations appears to have solved it, at least when using Inkscape's icon preview. Please, let me know how it looks on your end.

q4osteam commented:
The computer icon looks fine now. However it would need update for 16px size only as certain system icons gets different coloring for 16px size, see folder-16px icon for comparison:
![c1_folder16](https://github.com/user-attachments/assets/75b6fb07-875c-427f-85da-636466476496)
![c2_comp16](https://github.com/user-attachments/assets/69a02861-20b2-4bcd-9608-76cf4547bbae)

nm_wired_device still looks somewhat unsightly compared to its .png original:
orig png:
![b1_nmdev_orig](https://github.com/user-attachments/assets/aca4e9b5-3f94-49b6-b6db-4463928cdf46)
new svg:
![b2_nmdev_new](https://github.com/user-attachments/assets/e301d2ce-b4ef-4e44-9e59-9de9e2c9dff3)

The default desktop look
![avbox](https://github.com/user-attachments/assets/a7e6b80e-84a6-43a6-a613-57c3d6f10883)

Are you able to compile packages and check iconset look in a virtual machine ?

jaerrib committed:
Update to match other system icon colors

jaerrib commented:
I updated the size computer icon; hopefully that fits better with the rest of the theme.

As for the nm_wired_device, I'm a bit stumped as to how fix it. Is it out of the question to default to the one used in Papirus since we import quite a few of their icons already?

When I run 99_generate_iconset.sh, I get the following and wasn't quite sure what to do:

```bash
Before running hook1_prepare_icons > working ...
Running hook: /tmp/.00_gendeb_iconsname//hook1_prepare_icons.sh
Hook1 start.
* Using Papirus icons commit: "Waterleaf stable"

Before upstream copy > working ...
rsync: [sender] change_dir "/home/jaerrib/waterleaf-icon-theme/../.clone.tmp" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1338) [sender=3.2.7]

Before git checkouts > working ...
99_generate_iconset.sh: 34: cd: can't cd to /tmp/.00_gendeb_iconsname//papirus-folders/
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
99_generate_iconset.sh: 36: cd: can't cd to /tmp/.00_gendeb_iconsname//papirus-icon-theme/
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

Before patch > working ...
99_generate_iconset.sh: 42: cd: can't cd to /tmp/.00_gendeb_iconsname//papirus-icon-theme/
can't find file to patch at input line 4
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff -uNr papirus-icon-theme.a/tools/build_color_folders.sh papirus-icon-theme.b/tools/build_color_folders.sh
|--- papirus-icon-theme.a/tools/build_color_folders.sh  2020-07-19 13:23:53.780661935 +0200
|+++ papirus-icon-theme.b/tools/build_color_folders.sh  2020-07-19 13:46:29.000000000 +0200
--------------------------
File to patch:                                      
```

q4osteam commented:
Yes, the script has an issue, you only need to clone Papirus iconset first. Do that this way:
$ cd <some_path>/waterleaf-icon-theme/../
$ mkdir .clone.tmp
$ cd .clone.tmp/
$ git clone https://github.com/PapirusDevelopmentTeam/papirus-folders.git
$ git clone https://github.com/PapirusDevelopmentTeam/papirus-icon-theme.git
$ cd ../waterleaf-icon-theme/
and run the script:
$ sh 99_generate_iconset.sh
Ignore a few "scour" python xml parser errors safely. Does the script work now ?

We will fix the script asap and put a pull request.

q4osteam commented:
Computer icon looks fine now

q4osteam commented:
Nm_device_wired - we would prefer not to receive Papirus variant as the original png icon looks much better in Debonaire theme. Would it be possible to create as much as accurate copy png > svg ? for all sizes.

The only requirement is to follow the Waterleaf guidelines. The svg variant has a bit of an inaccurate proportion. It seems small and distorted, not as polished as the original png on the overall desktop look.

See for example 24x24 png/svg comparison, they are similar, but not the same:
png:
![b1_nmdev_orig](https://github.com/user-attachments/assets/0738ef9f-2870-493a-b773-0e2a1202be71)
svg:
![b2_nmdev_new](https://github.com/user-attachments/assets/46075aa2-26fd-45df-8ff5-b1afded9ea24)

jaerrib committed:
Redraw icons to better match original png versions

jaerrib commented:
I used the original png files as templates and redrew all 6 sizes for nm-device-wired in Inkscape.

q4osteam commented:
"The only requirement is to follow the Waterleaf guidelines."
The icons look much better now, but they still need modifications to follow guidelines before merging, compare for example to the 16x16 "blockdevice" svg icon.

blockdevice-16x16 (correct)
![blockdevice16](https://github.com/user-attachments/assets/59e271bf-ef31-45d2-9194-7cf2d1c44fdd)

nm_device_wired-16x16 (incorrect)
![nmdevicewired16](https://github.com/user-attachments/assets/11c1bb00-7d67-46e8-81ba-922659f7ac40)

16x16 icon should use 100% of the canvas size, so it should reach either horizontal or vertical edge of the canvas, or both of them. Any svg icon should also be perfectly horizontally as well as vertically centered.

jaerrib committed:
Center all icons, increase size of 16 icon to fit width

jaerrib commented:
Updated per feedback

q4osteam commented:
Well, nm_device_wired-16x16 is fine now :)

However nm_device_wired-32x32 doesn't meet the guidelines https://htmlpreview.github.io/?https://github.com/jaerrib/waterleaf-icon-theme/blob/development/dev_guide/waterleaf_guide.html

nm_device_wired-32x32:
![nm_device_wired-32x32](https://github.com/user-attachments/assets/c4b06ad5-3798-4ca4-a6e7-3c1520dc6c1d)

Please check whitespace, baseline, x-height.
Please check all the new icons to follow the guidelines properly.

jaerrib commented:
> Any svg icon should also be perfectly horizontally as well as vertically centered.

> nm_device_wired-32x32 doesn't meet the guidelines 
> ...
> Please check whitespace, baseline, x-height.

I'm confused as to what I'm targeting with the 32z32 icon. The guideline states a target of a baseline of 2 and an x-height of 28. The icon in the last commit had a baseline of 1.579 (which would be a little low, but the icon is centered vertically as per the earlier comment) and the x-height is 28.842 (slightly large).

Would you like the icon reduced in size? adjusted vertically? both?

q4osteam commented:
nm_device_wired-32x32 should meet by the guidelines:
whitespace: 2 (at least, depending on the drawing aspect ratio)
baseline: 2 (at least, depending on the drawing aspect ratio)
x-height: 28 (at most, depending on the drawing aspect ratio)
but it actually has:
whitespace:  less then 1 - should be: 2 (at least, depending on the drawing aspect ratio)
baseline: 1.579 - should be: 2 (at least, depending on the drawing aspect ratio)
x-height: 28.842 - should be: 28 (at most, depending on the drawing aspect ratio)
so it doesn't meet the guidelines.

Yes, the drawing should be scaled proportionally and centered on both the vertical and horizontal axes.

It's important, all the new icons and variants follow the guidelines. Yes, there could be exceptions as mentioned in the guidelines, but it's not subject of nm_device_wired.

jaerrib committed:
Reduce scale, adjust baseline height

q4osteam commented:
Additional valid rules by the guidelines:
1. All drawings are centered on the vertical axis for all sizes.
2. Icon sizes 16.24.32 are also centered on the horizontal axis because:  baseline == canvas_size - x-height - baseline
3. Icon sizes 48.64.128 are not centered on the horizontal axis because:  baseline != canvas_size - x-height - baseline

jaerrib commented:
Updated per feedback

q4osteam commented:
Well, on the example of nm_device_wired-32x32 we can see, we need to make guidelines more accurate in case the height of the drawing is smaller then its width.
Option 1: strictly force baseline to the value defined.
Option 2: center the drawing on the horizontal axis, so it will be set higher then the baseline, depending on the aspect ratio.
What would you mean would be better choice ?
We would rather lean to Op.1, but not sure enough at the moment. As soon as we arrange this, we can continue to review this pull request.

jaerrib commented:
I would lean towards option 1 as well so that it would be aligned similarly to other icons when it appears in a horizontal row, such as on the taskbar.

q4osteam commented:
After discussion within our team, we would definitely prefer option 1 as the default, with possible exceptions that would be in line with option 2. We will open a new issue to update the guidelines with a more detailed description in that regard. Now back to this PR.

nm_device_wired icon:
1. We would prefer nm_device_wired would be such an exception, please keep it in line with option 2. So all the nm_device_wired variants need to be centered on the horizontal axis.

2. We consider the nm_device_wired-16x16 complete and perfect, other variants should keep exactly the same aspect ratio, please check them too.

jaerrib committed:
Center icon vertically

jaerrib commented:
Vertically centered size 32 icon per feedback. Verified vertical positioning of all other sizes for nm_device_wired; no changes necessary.

q4osteam commented:
Now it looks better as centered. However there are still a few questions regarding to nm_device_wired.

- Aspect ratio
Should all variants within one icon keep the same aspect ratio ? We would think yes, if we want to get the best visual experience. nm_device_wired drawings doesn't follow fixed aspect ratio, for example:
nm_device_wired-24 .. 0.970
nm_device_wired-32 .. 1.053
If you would decide to remake drawings, we would suggest to set aspect ratio for nm_device_wired to 1. It would bring more simplicity and clear look.
What is your opinion on that ?

- Shadowing - note 1
Should we position 48+ drawings before adding shadows and let shadows overflow baseline, whitespace etc.. ? We would think yes, however it has consequence, so we would need reposition all shadowed drawings and let shadows overflow baseline, whitespace etc.. The aspect ratio would also be calculated before shadows addition.
Your opinion ?

- Shadowing - note 2
We have noticed some inconsistency in the current nm_device_wired shadows, as nm_device_wired-64 has much bolder shadow then nm_device_wired-128. Shadows for 48 variant should be subtle and become bolder as size increases.
Shouldn't that be the case ?

- Shadowing - note 3
Shadowing should be consistent across Waterleaf iconset.

We are sorry for complicating things, but trying to set clear and fixed rules to get as best as possible visual experience. We respect it's completely up to you when to merge this PR.

jaerrib commented:
> Now it looks better as centered. However there are still a few questions regarding to nm_device_wired.
> 
>     * Aspect ratio
>       Should all variants within one icon keep the same aspect ratio ? We would think yes, if we want to get the best visual experience. nm_device_wired drawings doesn't follow fixed aspect ratio, for example:
>       nm_device_wired-24 .. 0.970
>       nm_device_wired-32 .. 1.053
>       If you would decide to remake drawings, we would suggest to set aspect ratio for nm_device_wired to 1. It would bring more simplicity and clear look.
>       What is your opinion on that ?

Using one aspect across all the icon sizes would improve the overall consistency. Here's an example for the size 16 icon:

![snapshot1](https://github.com/user-attachments/assets/ae0a2e15-4820-4db9-92a3-ca7bd787c40a)

Larger sizes would, of course, incorporate the baseline and x-height but keep the 1:1 ratio for the drawn portion of the icon.


>     * Shadowing - note 1
>       Should we position 48+ drawings before adding shadows and let shadows overflow baseline, whitespace etc.. ? We would think yes, however it has consequence, so we would need reposition all shadowed drawings and let shadows overflow baseline, whitespace etc.. The aspect ratio would also be calculated before shadows addition.
>       Your opinion ?

I'm not opposed to either, but reworking all of the existing icons would be a fair amount of work. Are you just looking at this specific icon or everything 48+? (As an opportunity for improvement, this should probably be its own issue.)

>     * Shadowing - note 2
>       We have noticed some inconsistency in the current nm_device_wired shadows, as nm_device_wired-64 has much bolder shadow then nm_device_wired-128. Shadows for 48 variant should be subtle and become bolder as size increases.
>       Shouldn't that be the case ?

It's not that the shadow is necessarily darker but the width of the shadow relative to the icon size probably makes it appear more or less so. I can experiment with different options to see what works best.
 
>     * Shadowing - note 3
>       Shadowing should be consistent across Waterleaf iconset.

Agreed, but that should be part of another issue/PR. This one was really only about replacing the last of the png files.

> We are sorry for complicating things, but trying to set clear and fixed rules to get as best as possible visual experience. We respect it's completely up to you when to merge this PR.

Let's finalize the general look of the nm-device-wired icon with the 1:1 ratio to close out this PR and then move your suggestions into another issue to be worked on over time.

q4osteam commented:
1. Aspect ratio:
That sounds and looks good.

2. Shadowing - note 1:
All 48+, but we don't want to redo them right now. Just to follow this rule for new icons and occasionally fix the others step by step.

3. Shadowing - note 2:
Ok, we agree to move this on.

4. Shadowing - note 3:
It is intended more as a general rule.

We agree, so we will check the PR and post a feedback as soon as possible.

jaerrib committed:
Rework for consistent 1:1 aspect ratio

q4osteam commented:
We have tested the PR in action. For some reason you have changed nm_device_wired appearance. By our meaning it looks somewhat unsightly compared to its .png original. Baseline, whitespace, aspect ratio, shadowing and other aspects are now fine.

orig png:
![snapshot25](https://github.com/user-attachments/assets/38f9476f-0e5e-49e9-a790-149b3b6ed981)

new svg:
![snapshot26](https://github.com/user-attachments/assets/18928440-5402-4cd5-a0c0-824a95ec7638)

overall desktop look:
![VirtualBox_q4os-6](https://github.com/user-attachments/assets/bd62dd50-e755-4093-a984-7fe81064f2aa)

jaerrib commented:
I had made the changes to the appearance to achieve the 1:1 aspect ratio. The consequence was the narrowing of the "box" part of the arrows. The original png version is roughly 1.09:1 (slightly wider than it is tall). My response from two days shows the size 16 icon with the 1:1 ratio and the narrow sections. I was under the impression that you were okay with it, which is why I made all of the other sizes appear the same way.

I can attempt to finesse the dimensions to widen those portions of the icon to be more in line with the png version while maintaining the 1:1 ratio, but it may never quite look the same as the original. There will have to be a compromise _somewhere_. Which do you value more, the 1:1 aspect ratio or the thicker arrows?

q4osteam commented:
Nm_device_wired should maintain a 1:1 aspect ratio. Would it be possible to expand the arrows as indicated ? Both arrows should also be aligned and geometric.

![snapshot27](https://github.com/user-attachments/assets/4d27d823-b96a-42af-bcf7-f3072407a237)

We believe this change would almost preserve the perfect icon look and keep aspect ratio 1:1 at the same time.

jaerrib commented:
How's this? Still 1:1 aspect ratio but more in line with what you're looking for in terms of overall balance.

![snapshot2](https://github.com/user-attachments/assets/bcf7d618-12a2-444e-9653-5ce03edbff5d)

q4osteam commented:
Yep, that looks great :}

jaerrib commented:
> Yep, that looks great :}

Excellent. I will use that as the basis for the other sizes ~within the next day or two. (Depends on how much time I have after work today.)~ Done :heavy_check_mark:

jaerrib committed:
Thicken arrow while keeping 1:1 aspect ratio

q4osteam commented:
Great, nm_device_wired is now perfect :+1: What's next ? Do you want us to check other icons ?

jaerrib commented:
Go ahead and check the other icons associated with this PR if you want. If there are issues relevant to this specific PR, I can make those changes, but let's create new issues for any other problems we encounter. If everything is good, I will go ahead and merge this branch into development.

q4osteam commented:
Well, please check all the new icons by yourself to conform guidelines as best as possible and merge with the development branch. We will not review this PR more, but possibly create a a new issue. It would be great, if the new icons are polished as the old ones at least and meet the guidelines. Thanks for your work.
